src/Components/src/Microsoft.AspNetCore.Components/Rendering/Renderer.cs

```diff
@@ -295,5 +293,44 @@ private void RemoveEventHandlerIds(ArrayRange<int> eventHandlerIds, Task afterTa
                     RemoveEventHandlerIds(eventHandlerIdsClone, Task.CompletedTask));
             }
         }
+
+        /// <summary>
+        /// Releases all resources currently used by this <see cref="Renderer"/> instance.
+        /// </summary>
+        /// <param name="disposing"><see langword="true"/> if this method is being invoked by <see cref="IDisposable.Dispose"/>, otherwise <see langword="false"/>.</param>
+        protected virtual void Dispose(bool disposing)
+        {
+            List<Exception> exceptions = null;
+
+            foreach (var componentState in _componentStateById.Values)
+            {
+                if (componentState.Component is IDisposable disposable)
+                {
+                    try
+                    {
+                        disposable.Dispose();
+                    }
+                    catch (Exception exception)
+                    {
+                        // Capture exceptions thrown by individual components and rethrow as an aggregate.
+                        exceptions = exceptions ?? new List<Exception>();
+                        exceptions.Add(exception);
+                    }
+                }
+            }
+
+            if (exceptions != null)
+            {
+                throw new AggregateException(exceptions);
+            }
```

Comment:
Do we have an issue somewhere to clean this up? Throwing inside dispose isn't a nice thing to do, especially when you have inheritance involved. There are a bunch of places in this PR that will leak stuff on failed dispose 😆 

I realize that you didn't make it worse in this PR, but it is gross so I want to make sure it's on someone's plate to fix.

Reply:
Added it to this issue - https://github.com/aspnet/AspNetCore/issues/5476